### PR TITLE
[panw] Use high-res timestamp and ensure time zone config is applied

### PR DESCRIPTION
**Pull request by taylor-swanson**
## Proposed commit message

- The high-res timestamp will now be copied to `@timestamp`, if available, since this is a more accurate timestamp of when the event occurred.
- The tz_offset config value was only propagated to the pipeline if the logfile input was used. It is now propagated for the TCP and UDP inputs as well.
- If the tz_offset is not local, then it will override event.timezone, otherwise event.timezone will be set to the value given by the `add_locale` processor.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

```
cd packages/panw
elastic-package test
```

## Related issues

- Relates elastic/sdh-beats#4982

### CODE REVIEW

**Comment by taylor-swanson** [comment]
The fix is proving to be a bit more involved for the time zone configuration, and will require a bit more work before this PR is ready.

File: packages/panw/data_stream/panos/_dev/test/pipeline/test-panw-panos-globalprotect-sample.log-expected.json [hunk 1]
```diff
@@ -462,7 +462,7 @@
             ]
         },
         {
-            "@timestamp": "2021-03-02T09:55:39.000-05:00",
+            "@timestamp": "1969-12-31T19:00:00.000-05:00",
```
Comment (pkoutsovasilis) [inline packages/panw/data_stream/panos/_dev/test/pipeline/test-panw-panos-globalprotect-sample.log-expected.json:465]: hmmm this one seems a bit off as the year is not set in the log entry?!

File: packages/panw/data_stream/panos/_dev/test/pipeline/test-panw-panos-gtp-sample.log-expected.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
     "expected": [
         {
-            "@timestamp": "2019-10-09T10:20:15.000+07:00",
+            "@timestamp": "1970-01-01T07:00:00.000+07:00",
```
Comment (pkoutsovasilis) [inline packages/panw/data_stream/panos/_dev/test/pipeline/test-panw-panos-gtp-sample.log-expected.json:4]: can the remove check pick this up here as this starts with 1970?!

File: packages/panw/data_stream/panos/_dev/test/pipeline/test-panw-panos-globalprotect-sample.log-expected.json (hunk 1 above)
Comment (taylor-swanson) [reply packages/panw/data_stream/panos/_dev/test/pipeline/test-panw-panos-globalprotect-sample.log-expected.json:465]: Yeah so this is weird. The sample log uses a 1970 timestamp. Uh.. let me find the docs on this quick.

What I can't tell is if the 1970 date is something we made up in the example log or if it's the timestamp Palo Alot said to ignore (just with a different time zone)...

Comment (taylor-swanson) [reply packages/panw/data_stream/panos/_dev/test/pipeline/test-panw-panos-globalprotect-sample.log-expected.json:465]: Screenshot 2024-07-29 at 12 06 36 PM

Comment (pkoutsovasilis) [reply packages/panw/data_stream/panos/_dev/test/pipeline/test-panw-panos-globalprotect-sample.log-expected.json:465]: I am not sure either but my guess is that such timestamps found in log `Nov 30 16:09:08` they don't have a year thus they somehow not parsed at all and you fallback to the start of epoch which is something 1970?! then it applies the time zone and in some cases this throws you back to 1969-12-31 (amazing 🍷 year) or keeps you in 1970-01-01?!

Comment (taylor-swanson) [reply packages/panw/data_stream/panos/_dev/test/pipeline/test-panw-panos-globalprotect-sample.log-expected.json]: Regardless, I'll include 1970 as well, just to be safe with timezones.

**Review by pkoutsovasilis — APPROVED** [review]
LGTM

**Pull request merged**